### PR TITLE
Case insensitive like operator

### PR DESCRIPTION
**Pull request by Shredder121**

### CODE REVIEW

File: querydsl-core/src/main/java/com/querydsl/core/types/JavaTemplates.java [hunk 1]
```diff
@@ -58,7 +58,7 @@ public JavaTemplates() {
         add(Ops.TRIM, "{0}.trim()", Precedence.DOT);
         add(Ops.UPPER, "{0}.toUpperCase()", Precedence.DOT);
         add(Ops.MATCHES, "{0}.matches({1})");
-        add(Ops.MATCHES_IC, "{0}.matches({1})");
+        add(Ops.MATCHES_IC, "{0}.matches({1l})");
```
Comment (timowest) [inline querydsl-core/src/main/java/com/querydsl/core/types/JavaTemplates.java]: lowercase transformation also for the first arg?

File: querydsl-jdo/src/main/java/com/querydsl/jdo/JDOQLSerializer.java [hunk 2]
```diff
@@ -281,11 +282,17 @@ protected void visitOperation(Class<?> type, Operator operator, List<? extends E
...
 
-        } else if (operator == Ops.LIKE || operator == Ops.LIKE_ESCAPE) {
+        } else if (operator == Ops.LIKE || operator == Ops.LIKE_ESCAPE || operator == Ops.LIKE_IC || operator == Ops.LIKE_ESCAPE_IC) {
             @SuppressWarnings("unchecked") //This is the expected type for like
-            Expression<String> rightArg = (Expression<String>) args.get(1);
+            Expression<String> string = (Expression<String>) args.get(0);
+            @SuppressWarnings("unchecked") //This is the expected type for like
+            Expression<String> regex = ExpressionUtils.likeToRegex((Expression<String>) args.get(1), false);
+            if (operator == Ops.LIKE_IC || operator == Ops.LIKE_ESCAPE_IC) {
+                string = Expressions.stringOperation(Ops.LOWER, string);
```
Comment (timowest) [inline querydsl-jdo/src/main/java/com/querydsl/jdo/JDOQLSerializer.java]: could you do the lowercase transformation in such a way that it transforms constants by lowercasing the constant string and wrapping it into a constant?
other expressions can be wrapped into an operation.
Maybe utility methods for that into ExpressionUtils?

File: querydsl-sql/src/main/java/com/querydsl/sql/DB2Templates.java [hunk 3]
```diff
@@ -149,10 +146,8 @@ public DB2Templates(char escape, boolean quote) {
         setPrecedence(Precedence.COMPARISON, Ops.IS_NULL, Ops.IS_NOT_NULL, Ops.LIKE, Ops.LIKE_ESCAPE, Ops.BETWEEN,
                 Ops.IN, Ops.NOT_IN, Ops.EXISTS);
 
-        // other like cases
-        setPrecedence(Precedence.COMPARISON, Ops.ENDS_WITH, Ops.ENDS_WITH_IC,
-                Ops.STARTS_WITH, Ops.STARTS_WITH_IC,
-                Ops.STRING_CONTAINS, Ops.STRING_CONTAINS_IC);
+        setPrecedence(Precedence.COMPARISON,
```
Comment (timowest) [inline querydsl-sql/src/main/java/com/querydsl/sql/DB2Templates.java]: maybe overload with Collection?

Comment (Shredder121) [inline querydsl-sql/src/main/java/com/querydsl/sql/DB2Templates.java]: I was also thinking of that, I agree with that.

File: querydsl-jdo/src/main/java/com/querydsl/jdo/JDOQLSerializer.java (hunk 2 above)
Comment (Shredder121) [inline querydsl-jdo/src/main/java/com/querydsl/jdo/JDOQLSerializer.java]: Please advise, the `String.toLowerCase` transormation with or without `Locale`?
Query parameters are not necessarily in English too.

Comment (timowest) [inline querydsl-jdo/src/main/java/com/querydsl/jdo/JDOQLSerializer.java]: Locale.ENGLISH for now,and a general fix of locale handling in Querydsl could be the topic of another PR.

Comment (Shredder121) [inline querydsl-jdo/src/main/java/com/querydsl/jdo/JDOQLSerializer.java]: Okay then.

**Comment by Shredder121** [comment]
JPQLTemplates still needs to be addressed.

**Comment by timowest** [comment]
Looks good

File: querydsl-sql/src/test/java/com/querydsl/sql/suites/HsqldbLiteralsSuiteTest.java [hunk 4]
```diff
@@ -3,10 +3,10 @@
 import org.junit.BeforeClass;
 import org.junit.experimental.categories.Category;
 
-import com.querydsl.core.testutil.H2;
+import com.querydsl.core.testutil.HSQLDB;
 import com.querydsl.sql.*;
 
-@Category(H2.class)
+@Category(HSQLDB.class)
```
Comment (johnktims) [inline querydsl-sql/src/test/java/com/querydsl/sql/suites/HsqldbLiteralsSuiteTest.java]: Good eye, Ruben!

**Comment by johnktims** [comment]
Looks good to me.
